src/pages/workspaces/workspace/Data.js

```diff
@@ -322,25 +335,45 @@ const WorkspaceData = _.flow(
     )
   }
 
-  const loadMetadata = withErrorReporting('Error loading workspace entity data', async () => {
-    const [entityMetadata, { resources: snapshotMetadata }] = await Promise.all([
-      Ajax(signal).Workspaces.workspace(namespace, name).entityMetadata(),
-      Ajax(signal).Workspaces.workspace(namespace, name).listSnapshot(1000, 0)
-    ])
-
-    const snapshotEntities = await Promise.all(_.map(({ name: snapshotName }) => {
-      return Ajax(signal).Workspaces.workspace(namespace, name).snapshotEntityMetadata(namespace, snapshotName)
-    }, snapshotMetadata))
+  const loadEntityMetadata = async () => {
+    try {
+      setEntityMetadataError(false)
+      const entityMetadata = await Ajax(signal).Workspaces.workspace(namespace, name).entityMetadata()
+
+      setSelectedDataType(getSelectionType() === 'entities' && !entityMetadata[selectedDataType] ? undefined : selectedDataType)
+      setEntityMetadata(entityMetadata)
+    } catch (error) {
+      reportError('Error loading workspace entity data', error)
+      setEntityMetadataError(true)
+      setSelectedDataType(undefined)
+      setEntityMetadata({})
+    }
+  }
 
-    const snapshotDetails = _.fromPairs(_.map(([entities, metadata]) => {
-      return [metadata.name, { entityMetadata: entities, id: metadata.reference.snapshot }]
-    },
-    _.zip(snapshotEntities, snapshotMetadata)))
+  const loadSnapshotMetadata = async () => {
+    try {
+      setSnapshotMetadataError(false)
+      const { resources: snapshotMetadata } = await Ajax(signal).Workspaces.workspace(namespace, name).listSnapshot(1000, 0)
+
+      const snapshotEntities = await Promise.all(_.map(({ name: snapshotName }) => {
+        return Ajax(signal).Workspaces.workspace(namespace, name).snapshotEntityMetadata(namespace, snapshotName)
+      }, snapshotMetadata))
```

Comment:
for this PR or a future one: can these be loaded lazily? Do we really have to proactively query for every single snapshot reference when the the Data tab is first opened? For a user with multiple snapshots, this could quickly become onerous, and involve a lot of queries that never end up getting  used.

Reply:
Definitely, we don't have to load the tables in a snapshot before someone expands it.

Reply:
I'd want to save that for a separate PR, though.